### PR TITLE
Update docs links blocks documentation

### PR DESCRIPTION
**Pull request by Brianmitchtay**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

We've recently refactored and updated our blocks & store editing documentation. This PR updates the in-product links to point to the new documentation.

I performed a search, and replaced all found instances of https://woocommerce.com/document/cart-checkout-blocks-status/ with links to the appropriate section of new documentation on https://woocommerce.com/document/woocommerce-store-editing/customizing-cart-and-checkout/

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Confirm the link in the "Classic Shortcode" notice points to the new docs & that the link works**
1. Open playground of this PR
2. Head to the site editor and the Cart template. 
3. Swap the cart block for the "classic shortcode" using the block transform in the block toolbar
4. Confirm the "In our documentation" link in the notice points to `https://woocommerce.com/document/woocommerce-store-editing/customizing-cart-and-checkout/`
5. Click the link and confirm that you arrive at the correct page
6. Repeat steps 2-5 for the same notice on the Checkout Block

**Confirm that the "Learn More" link in the checkout block incompatibility notice points to the new docs & that the link works**
1. Install and activate an extension/payment gateway which has declared incompatibility with the block.
2. Return to the checkout template in the site editor. 
3. Select the Checkout block and expand the settings sidebar.
4. Confirm that the "Learn More" link in the incompatibility notice points to `https://woocommerce.com/document/woocommerce-store-editing/customizing-cart-and-checkout/#incompatible-extensions`
5. Click the link and confirm that you arrive at the correct page

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
Updating documentation links, I don't believe a comment is necessary for this small change.
</details>

### CODE REVIEW

**Review by mikejolley — APPROVED** [review]
CI is green this is good to go 👍🏻

File: plugins/woocommerce/changelog/50239-update-in-product-blocks-docs-links [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: tweak
+Comment: Updating documentation links, I don't believe a comment is necessary for this small change.
```
Comment (mikejolley) [inline plugins/woocommerce/changelog/50239-update-in-product-blocks-docs-links:3]: Was this added before you checked the " This Pull Request does not require a changelog entry. (Comment required below)" box? 

If so I think this should be removed before merging.

Comment (Brianmitchtay) [reply plugins/woocommerce/changelog/50239-update-in-product-blocks-docs-links:3]: I'm not sure how the changelog entries get added, it looks like it was added by an automation, however I did have the  "This Pull Request does not require a changelog entry. (Comment required below)" box checked before I published this PR yesterday, so not sure why it added the entry anyway. 

Let's remove it before merging 👍 I went ahead and deleted the file.

Comment (mikejolley) [reply plugins/woocommerce/changelog/50239-update-in-product-blocks-docs-links:3]: I asked in Slack and this is apparently expected, so leave it in place.

**Pull request merged**